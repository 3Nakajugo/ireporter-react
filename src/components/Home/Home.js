import React, { Component } from 'react';
import '../../assests/styles/home.css';

class Home extends Component {
  handleClick = (e) => {
    e.preventDefault();
    document.location.href = '/signup';
  };

  render() {
    return (
      <div>
        <div>
          <h1> iReporter</h1>
          <div className="intro">
            <p>
              iReporter is an online platform that enables any/every citizen to
              <br />
              bring any form of corruption to the notice of appropriate authorities and the
                    general public.
            </p>
            <div className="index-btn">
              <form>
                <div className="start-btn">
                  <input type="button" value="Get Started" onClick={this.handleClick} />
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
export default Home;
