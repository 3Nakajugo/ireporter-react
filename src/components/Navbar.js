import React, { Component } from 'react'
import { Link } from "react-router-dom";
import '../assests/styles/nav.css'

class Navbar extends Component {
  render() {
    return (
      <div>
        <nav className="navbar navbar-expand-lg navbar-dark">
          <button className="navbar-toggler nav-button" type="button"
            data-toggle="collapse" data-target="#navbarTogglerah"
            aria-controls="navbarTogglerDemo01" aria-expanded="false"
            aria-label="Toggle navigation">
            <span className="navbar-toggler-icon" /></button>
          <div className="collapse navbar-collapse" id="navbarTogglerah">
            <ul className="navbar-nav mr-auto mt-2 mt-lg-0">
              <li className="nav-item active"></li>
            </ul>
            <ul className="navbar-nav ml-2 ">
              <li className="nav-item"><Link className="navlink" to = "/">Home</Link></li>
              <li className="nav-item ml-4">Login</li>
            </ul>
          </div>
        </nav >
      </div>
    )
  }
}

export default Navbar;