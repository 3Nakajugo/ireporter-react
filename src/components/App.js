import React, { Component } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';
import { ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import store from '../redux/store';
import Home from './Home/Home';
import Login from './Login/login';
import Signup from '../containers/signupContainer';

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <BrowserRouter>
        <ToastContainer />
          <Switch>
            <Route exact path="/" component={Home} />
            <Route path="/login" component={Login} />
            <Route path="/signup" component={Signup} />
          </Switch>
        </BrowserRouter>
      </Provider>
    );
  }
}

export default App;
