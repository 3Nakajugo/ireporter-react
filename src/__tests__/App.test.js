import React from 'react';
import { shallow } from 'enzyme';
import App from '../components/App';
import Home from '../components/Home/Home';
import Navbar from '../components/Navbar';

describe('<App />', () => {
  it('it should render the app', () => {
    const wrapper = shallow(<App />);
    const homewrapper = shallow(<Home />);
    const navwrapper = shallow(<Navbar />);
    expect(wrapper).toMatchSnapshot();
    expect(homewrapper).toMatchSnapshot();
    expect(navwrapper).toMatchSnapshot();
  });
});
