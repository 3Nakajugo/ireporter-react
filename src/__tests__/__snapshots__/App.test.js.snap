// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> it should render the app 1`] = `ShallowWrapper {}`;

exports[`<App /> it should render the app 2`] = `ShallowWrapper {}`;

exports[`<App /> it should render the app 3`] = `ShallowWrapper {}`;

exports[`<App /> it should render the app 4`] = `ShallowWrapper {}`;
